### DISCUSSION TITLE
Sorting pages from nav.pages list

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#3377] [Q&A] [answered]
Using theme extension and jina2 templating, I try to sort pages by their date but so far had no real luck with it.

I first tried `{% for page in nav.pages|sort(attribute="date") %}` but it seems like the sort attribute has no impact on the result, no matter what I try here...
I also tried to use `meta.` with values, but this only causes `UndefinedError`s, most likely because meta is not part of the nav? Although later on can I extract and use meta values from an individual page entity....

I'm generally confused and hope there is some way to alter the sorting of nav items...

**Comment by pawamoy** [comment]
Maybe some of your pages do not define a front-matter, or a date within it, therefore some pages `meta` or `meta.date` attribute are undefined (in Jinja terms). You can filter such pages out before sorting with `nav.pages|rejectattr("meta", none)|rejectattr("meta.date", none)`, or something similar.

**Reply by Andre601** [reply]
Ah, okay. Will try that.
And yes, the index page is one that doesn't have a date metadata set.

**Reply by Andre601** [reply]
Tried it and still the same issue.

I've tried `{% for page in nav.pages|rejectattr("meta", "none")|rejectattr("meta.date", "none")|sort("meta.date") %}` but I get `jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'date'`

**Reply by Andre601** [reply]
Forgot the attribute=...
Will test again

Edit: No change at all.

**Reply by pawamoy** [reply, marked as answer]
Try "undefined" instead of "none".

**Reply by Andre601** [reply]
Tried it with `sort(attribute="meta.date")` and `sort(attribute="date")`

First one still returns the same error as mentioned above while second one changes nothing (Most likely because page doesn't have a direct date attribute so it treats it as empty)

**Reply by Andre601** [reply]
> Try "undefined" instead of "none".

This finally worked. Thank you.
